keel-plugin/src/main/kotlin/com/netflix/spinnaker/keel/plugin/ResourceHandler.kt

```diff
@@ -83,6 +83,17 @@ interface ResourceHandler<T : Any> : KeelPlugin {
 }
 
 /**
- * Represents some kind of conflict that prevents the plugin from interacting with the resource.
+ * Searches a list of `ResourceHandler`s and returns the first that supports [apiVersion] and
+ * [kind].
+ *
+ * @throws UnsupportedKind if no appropriate handlers are found in the list.
  */
-class ResourceConflict(message: String) : RuntimeException(message)
+internal fun List<ResourceHandler<*>>.supporting(
+  apiVersion: ApiVersion,
+  kind: String
+): ResourceHandler<*> =
+  find { it.apiVersion == apiVersion && it.supportedKind.first.singular == kind }
+    ?: throw UnsupportedKind(apiVersion, kind)
```

Comment:
Previously we were returning `null` and then the caller threw this exception. I figured it made more sense to just do it here so the caller can assume a non-null return.